server/text_generation_server/utils/layers.py

```diff
@@ -248,6 +249,19 @@ def get_linear(weight, bias, quantize):
                 bits,
                 groupsize,
             )
+    elif quantize == "awq":
+        try:
+            qweight, qzeros, scales, bits, groupsize = weight
+        except Exception:
+            raise NotImplementedError(
+                f"The passed weight is not `awq` compatible, loader needs to be updated."
+            )
+        in_features = qweight.shape[0]
+        out_features = qweight.shape[1] * 32 // bits
+        linear = WQLinear(w_bit=bits, group_size=groupsize, in_features=in_features, out_features=out_features, bias=bias is not None, dev=qweight.device)
```

Comment:
Can we modify the init logit so the allocation doesn't have to happen ?

WQLinear(qweight, qzeros, scales, groupsize, bias)

And modify `__init__` such that it infers `in_features` and `out_features` directly ?

It's should lead to simpler code, avoid odd behavior of `register_buffer`, and be more in line with what we already have done.

Reply:
@Narsil: This is fixed.